mlos_bench/mlos_bench/services/remote/azure/azure_vm_services.py

```diff
@@ -191,8 +192,11 @@ def __init__(self,
         assert template is not None and isinstance(template, dict)
         self._deploy_template = template
 
-        self._deploy_params = merge_parameters(
-            dest=self.config['deploymentTemplateParameters'].copy(), source=global_config)
+        # Allow for recursive variable expansion as we do with global params and const_args.
+        deploy_params = DictTemplater(self.config['deploymentTemplateParameters'].copy()).expand_vars(
+            extra_source_dict=global_config or {})
```

Comment:
I don't think there is a need to copy, so
```suggestion
        deploy_params = DictTemplater(self.config['deploymentTemplateParameters']).expand_vars(
            extra_source_dict=global_config)
```